### PR TITLE
Document that APIs may throw IOException

### PR DESCRIPTION
**Pull request by tmat**
And convert UnauthorizedAccessException to IOException.

### CODE REVIEW

**Comment by tmat** [comment]
@nguerrera

**Comment by nguerrera** [comment]
:+1:

**Comment by stephentoub** [comment]
LGTM

File: src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryMapLightUp.cs [hunk 1]
```diff
@@ -217,6 +217,10 @@ internal static IDisposable CreateViewAccessor(object memoryMap, long start, int
                 s_lazyIsAvailable = false;
                 return null;
             }
+            catch (TargetInvocationException ex) when (ex.InnerException is UnauthorizedAccessException)
+            {
+                throw new IOException(ex.InnerException.Message, ex.InnerException);
+            }
```
Comment (stephentoub) [inline src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryMapLightUp.cs]: Does this exception changing and no tests changing indicate that we're missing tests?

Comment (tmat) [inline src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryMapLightUp.cs]: We don't test all possible exceptions coming from memory map currently. We would need to add test hooks to emulate throwing these exceptions.

**Pull request merged**